pkg/syncer/metadatasyncer.go

```diff
@@ -1391,6 +1396,12 @@ func podAdded(obj interface{}, metadataSyncer *metadataSyncInformer) {
 	if metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) &&
 		metadataSyncer.clusterFlavor != cnstypes.CnsClusterFlavorWorkload &&
 		metadataSyncer.clusterFlavor != cnstypes.CnsClusterFlavorGuest {
+
+		if len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
```

Comment:
This check in not required.
We call volumeMigrationService methods only when PV is in-tree vSphere volume.
which will not be the case for multi-VC deployment.

Also, `initVolumeMigrationService` also has this protection.

```golang
if !isMultiVCenterFssEnabled {
		volManager = metadataSyncer.volumeManager
	} else {
		if len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
			// Migration feature switch is enabled and multi vCenter feature is enabled, and
			// Kubernetes Cluster is spread on multiple vCenter Servers.
			return logger.LogNewErrorf(log,
				"volume-migration feature is not supported on Multi-vCenter deployment")
		}
```